### PR TITLE
Add on-demand bluemix login support

### PR DESCRIPTION
**Pull request by raymondfeng**
### Description

Use API to log in on demand

#### Related issues

- strongloop-internal/scrum-loopback#1536

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by hacksparrow** [comment]
@raymondfeng what about unit tests?

**Comment by hacksparrow** [comment]
@raymondfeng Prompts for "autoscaling" and "appmetrics" are omitted if prompt for "manifest.yml" is answered "n". "autoscaling" and "appmetrics" are independent of "manifest.yml".

Apart from that, loved the feature and implementation!

**Comment by hacksparrow** [comment]
I will list some workflow and the corresponding bugs below.

**Comment by hacksparrow** [comment]
```sh
$ lb bluemix --login
Log into your Bluemix account:
? URL for bluemix APIs: https://api.ng.bluemix.net
? Use Single Sign-On (SSO): No
? Email for your account: yaapahage@yahoo.com
? Password for your account: *********
Logging into Bluemix...
Listing organizations from Bluemix...
> Default organization: yaapahage
Listing spaces from Bluemix...
> Default space: demos
? Remember my login: Yes
(node:7895) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'guid' of undefined
(node:7895) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by hacksparrow** [comment]
```
$ lb bluemix --sso
Log into your Bluemix account:
? URL for bluemix APIs: https://api.ng.bluemix.net
? Get one time passcode from https://iam.ng.bluemix.net/oidc/passcode: ******
Logging into Bluemix...
Listing organizations from Bluemix...
events.js:161
      throw er; // Unhandled 'error' event
      ^

Error: Invalid Auth Token
    at Request._callback (/Users/work/projects/generator-loopback/node_modules/loopback-bluemix/lib/cf.js:104:13)
    at Request.self.callback (/Users/work/projects/generator-loopback/node_modules/request/request.js:188:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:192:7)
    at Request.<anonymous> (/Users/work/projects/generator-loopback/node_modules/request/request.js:1171:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:189:7)
    at IncomingMessage.<anonymous> (/Users/work/projects/generator-loopback/node_modules/request/request.js:1091:12)
    at Object.onceWrapper (events.js:291:19)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:186:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
```

**Comment by hacksparrow** [comment]
Should be able to execute `lb bluemix --login` and `lb bluemix --sso` anywhere on the file system, currently it throws if not executed in a LoopBack dir.

File: test/bluemix.test.js [hunk 1]
```diff
@@ -134,6 +138,43 @@ describe('loopback:bluemix generator', function() {
       done();
     });
   });
+
+  it('should login with user/password', function(done) {
+    if (!process.env.BLUEMIX_EMAIL) {
+      console.error('Missing BLUEMIX_EMAIL and BLUEMIX_PASSWORD env vars');
+      return done();
```
Comment (hacksparrow) [inline test/bluemix.test.js]: This makes the test pass anyway, will fix it. Applies to 'should login with SSO passcode' too.

**Comment by raymondfeng** [comment]
@hacksparrow Please feel free to merge when you are ready.

**Comment by hacksparrow** [comment]
@raymondfeng thanks, I will.

**Comment by hacksparrow** [comment]
@raymondfeng  some checks are failing with "No identified problem", and another ([cis-jenkins] PR Builder) is failing with "npm ERR! 404  'fd-slicer' is not in the npm registry". [`fd-slicer` actually exists on npm](https://www.npmjs.com/package/fd-slicer). All tests are passing locally, though.

All planned work have been completed, the commits squashed. I need your confirmation before merging, or the plan of action to handle these strange errors.

**Comment by hacksparrow** [comment]
Thanks @raymondfeng

**Pull request merged**